src/schema/users.ts

```diff
@@ -850,14 +852,56 @@ export const resolvers: IResolvers<any, Context> = {
         .orderBy('date')
         .getRawMany();
     },
-    userStreak: (_, __, ctx: Context, info): Promise<GQLUserStreak> =>
-      graphorm.queryOneOrFail<GQLUserStreak>(ctx, info, (builder) => ({
-        queryBuilder: builder.queryBuilder.where(
-          `"${builder.alias}"."userId" = :id`,
-          { id: ctx.userId },
-        ),
-        ...builder,
-      })),
+    userStreak: async (_, __, ctx: Context, info): Promise<GQLUserStreak> => {
+      const streak = await ctx.con
+        .getRepository(UserStreak)
+        .createQueryBuilder()
+        .select('"lastViewAt"') // needs to consider user's timezone
+        .addSelect('"userId"')
+        .getRawOne();
+      const getStreak = () => getUserStreak(ctx, info, ctx.userId);
+      const lastView = await getStreakLastView(ctx.con, streak);
+
+      if (!lastView) {
+        return getStreak();
+      }
+
+      const today = new Date();
+      const dayToday = today.getDay();
+      const difference = differenceInDays(today, lastView);
+
+      if (difference === 0) {
+        return getStreak();
+      }
+
+      // Even though it is the weekend, we should still clear the streak for when the user's last read was Thursday
+      // Due to the fact that when Monday comes, we will clear it anyway when we notice the gap in Friday
```

Comment:
Is it not "free" but could still show as "Read" if you decided to read?
(not really sure where this part is used, so more of a question at this point)

Reply:
Yes, if you read anything, it will count. The concern here is that, let's say today is Saturday, and the user's last read was Thursday. We should clear it since there was a gap, which is the Friday.